src/routes.js

```diff
@@ -1,32 +1,77 @@
-import React from 'react';
+/* eslint-disable */
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { TransitionGroup, CSSTransition } from 'react-transition-group';
 import { Route, Redirect, Switch } from 'react-router-dom';
 import { Protocol, Start } from './containers';
 import { getCSSVariableAsNumber } from './utils/CSSVariables';
+import tween from './behaviours/Tweened/tween';
 
-const routes = ({ location }) => (
-  <TransitionGroup component={null}>
-    <CSSTransition appear timeout={getCSSVariableAsNumber('--animation-duration-fast-ms')} classNames="route" key={location.key}>
-      <Switch location={location}>
-        <Route
-          exact
-          path="/edit/:protocol"
-          component={Protocol}
-        />
-        <Route
-          exact
-          path="/"
-          component={Start}
-        />
-        <Redirect to="/" />
-      </Switch>
-    </CSSTransition>
-  </TransitionGroup>
-);
+const dispatchRouteAnimations = ({ pathname }) => {
+  switch (true) {
+    case /^\/$/.test(pathname):
+      tween({ name: "foo", from: "overview-panel", to: "protocol-stack" });
```

Comment:
The tween back to the protocol stack is interesting, since the click area includes the titles below (and margin around) the image, but you'd probably just want to tween to the image. Is this flexible enough to delegate a different view object to transition or does that throw a wrench in things?

Reply:
That's an interesting point: we have references to both from and to elements, we could clone them both!

Probably the way I'd implement that is that the overview panel would fade to a solid color as it moves over the top of where the protocol stack will appear, once there it fades out to reveal the stack underneath, which could fade in titles at the same time - this is part of the issue I have with this implementation, all the synchrony required.